python/cudf/cudf/core/column/numerical.py

```diff
@@ -276,7 +277,12 @@ def _binaryop(self, other: ColumnBinaryOperand, op: str) -> ColumnBase:
                     f"{other_cudf_dtype.type.__name__}"
                 )
             if self.dtype.kind == "b" or other_cudf_dtype.kind == "b":
-                out_dtype = np.dtype(np.bool_)
+                if cudf.get_option("mode.pandas_compatible"):
```

Comment:
Does this mean we have an intended difference in behavior (whether Kleene logic is used) depending on `mode.pandas_compatible`? I am inclined to say we should always align with pandas-compatible behavior here since there should be no performance loss from matching the upstream behavior.

Another option would be to deprecate the current behavior and note that a future cudf version will change to match pandas.

Reply:
Updated the code to be not pandas compatibility mode dependent.